src/lib/components/image/Image.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState } from "react";
+import classNames from "classnames";
+import { BiShow, BiError } from "react-icons/bi";
+import { VuiFlexContainer } from "../flex/FlexContainer";
+import { VuiFlexItem } from "../flex/FlexItem";
+import { VuiText } from "../typography/Text";
+import { VuiSkeleton } from "../skeleton/Skeleton";
+import { VuiImagePreview } from "./ImagePreview";
+import { VuiTooltip } from "../tooltip/Tooltip";
+import { VuiIcon } from "../icon/Icon";
+import { VuiTextColor } from "../typography/TextColor";
+
+export type ImageSize = "xs" | "s" | "m" | "l" | "xl" | "full";
+export type CaptionPosition = "top" | "bottom" | "overlay";
+
+type Props = {
+  src: string;
+  alt?: string;
+  caption?: string;
+  captionPosition?: CaptionPosition;
+  size?: ImageSize;
+  className?: string;
+  isLoading?: boolean;
+  allowPreview?: boolean;
+  errorMessage?: string;
+  previewTitle?: string;
+  previewDescription?: string;
+  previewCaption?: string;
+  controls?: {
+    allowRotation?: boolean;
+    allowZoom?: boolean;
+    allowReset?: boolean;
+    allowDrag?: boolean;
+  };
+};
+
+const sizeMap = {
+  xs: 100,
+  s: 200,
+  m: 400,
+  l: 600,
+  xl: 800,
+  full: "100%"
+} as const;
+
+const directionMap = {
+  top: "columnReverse",
+  bottom: "column"
+} as const;
+
+const getFlexDirection = (position: keyof typeof directionMap) => {
+  return directionMap[position];
+};
+
+const sizeConfig = {
+  xs: { captionSize: "xs", skeletonRows: 2.5, previewTextSize: "xs", previewIconSize: 16 },
+  s: { captionSize: "xs", skeletonRows: 2.5, previewTextSize: "s", previewIconSize: 16 },
+  m: { captionSize: "m", skeletonRows: 3.25, previewTextSize: "m", previewIconSize: 24 },
+  l: { captionSize: "m", skeletonRows: 3.25, previewTextSize: "m", previewIconSize: 24 },
+  xl: { captionSize: "l", skeletonRows: 4.25, previewTextSize: "l", previewIconSize: 32 },
+  full: { captionSize: "l", skeletonRows: 4.25, previewTextSize: "l", previewIconSize: 32 }
+} as const;
+
+export const VuiImage = ({
+  src,
+  alt,
+  caption,
+  captionPosition = "bottom",
```

Comment:
Can we remove this prop and always position the caption on the bottom? While I think all of the options are cool, introducing choice means that we as we developers sometimes make choices inconsistently throughout our UIs, which results in visual inconsistency for our users. I'd prefer to err on the side of too little choice than too much choice for the first iteration of a component, because it's typically easier to add props than it is to take them away. And once a the component is in use, it's easier to identify the props you need.